### PR TITLE
Calculate Hessian-vector product using 2nd derivative

### PR DESCRIPTION
**Pull request by mjacar**
Not all PyTorch operations are twice differentiable as of yet but I'll keep this PR open as a placeholder in the meantime.

### CODE REVIEW

**Comment by AjayTalati** [comment]
Hi @mjacar, a temporary work around for the second derivative of the softmax function has been suggested by @fmassa, [here](https://github.com/pytorch/pytorch/issues/440#issuecomment-306692826).

Maybe you want to give it a try?

**Comment by mjacar** [comment]
Already did this.
